gitRepo: 2.14.5 -> 2.15.3

otavio opened the pull request:
###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

otavio committed:
gitRepo: 2.14.5 -> 2.15.3

Signed-off-by: Otavio Salvador <otavio@ossystems.com.br>

r-rmcgibbo commented:
Result of `nixpkgs-review pr 124904` at 90f031ad run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>gitRepo</li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/applications/version-management/git-repo/default.nix:27:3:
```
   |
27 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/applications/version-management/git-repo/default.nix:16:15:
```
   |
16 |   patches = [ ./import-ssl-module.patch ];
   |               ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/applications/version-management/git-repo/default.nix:48:5:
```
   |
48 |     maintainers = [ ];
   |     ^
```
</li>
  </ul>
</details>

otavio committed:
gitRepo: add myself as maintainer

Signed-off-by: Otavio Salvador <otavio@ossystems.com.br>

otavio committed:
gitRepo: Drop unnecessary patch

The import-ssl-module.patch does not seem to be need and the tool seem
to be working just fine. Drop the patch.

Signed-off-by: Otavio Salvador <otavio@ossystems.com.br>

otavio committed:
gitRepo: add missing preInstall and postInstall hooks calls

Signed-off-by: Otavio Salvador <otavio@ossystems.com.br>

otavio committed:
gitRepo: Add updateScript support for easier upgrade in future

Signed-off-by: Otavio Salvador <otavio@ossystems.com.br>
